### DISCUSSION TITLE
Select similar sibling nodes.

### DISCUSSION THREAD

**Discussion by Qlio** [nvim-treesitter/nvim-treesitter#4339] [Q&A] [answered]
Hello,
1) I'm trying to customize by typescript folding query. I want to fold similar sibling nodes. For example, I want to fold all these imports into one fold:
```ts
import a from 'a';
import b from 'b';

import c from 'c';

// etc...
```

I tried to group them like this, but no luck:
```
(
  (import_statement)
  (import_statement)*
) @fold
```
2) Another question: Is it possible to include the following blank line in the fold?:
```ts
if (true) {
   // anything
}
                           // <---- I want this blank line to be included to the upper fold (if statements fold)
function doSomething() {
    // anything
}
```

**Comment by theHamsta** [comment, marked as answer]
At the moment you can only select individual nodes. There's a upstream PR that would allow to first iterate over all nodes mentioned `*`. Also the `fold` module would need to be made aware of this. Nvim-treesitter-textobects uses `make-range!` which can be used to create fake node objects that work with most tree-sitter modules. It's a bit a workaround that queries only select nodes but you can't describe ranges.

**Reply by theHamsta** [reply]
Hmm, it's possible that `make-range!` does work with the new query API of nvim

**Reply by GitMurf** [reply]
@theHamsta Are there any updates on this? I am trying to get single line comments to "group" together so when I comment out a section of code I can fold it all. It does not seem to work. I have tried all of the following:

```
(comment)+ @fold

(
  (comment)
  .
  (comment)+
)  @fold

; for testing 3 comments in a row
(
  (comment).(comment).(comment)
) @fold
```

I am trying to be able to fold things like this:

```lua
-- the chunk of comments below I want to be able to fold
local foo = "bar"

-- local bar = "baz"
-- local baz = "qux"
-- local qux = "help please!"

local failure = "fold the 3 comment lines above"
```

Thanks in advance!
